pkg/typesystem/typesystem.go

```diff
@@ -180,6 +180,31 @@ func (t *TypeSystem) IsDirectlyRelated(target *openfgapb.RelationReference, sour
 	return false
 }
 
+func (t *TypeSystem) HasTypeInfo(objectType, relation string) (bool, bool) {
```

Comment:
I don't love this function signature returning (bool, bool), but I kept the same pattern as we are using elsewhere. I'd like to revisit this in a subsequent PR and changes these back to appropriate errors.

So

```
func (t *TypeSystem) HasTypeInfo(objectType, relation string) (bool, error) {...}
```

WDYT?

Reply:
I'm unclear why it's not using just the left hand side and have a single one.

Reply:
Do you mean just return a single bool? I'm confused what you mean by 

> why it's not using just the left hand side and have a single one.

What are you referring to with left hand side and have a single one?

Reply:
> Do you mean just return a single bool?

I think so. I'm also confused because the function name is a question and it returns two answers :P

Reply:
It returns `(bool, bool)` because the `objectType` or `relation` may be undefined for the typesystem constructed from the authorization model.

For example,

```
type user

type document
    relations
      define viewer: [user]
```

```
hasTypeInfo, ok := HasTypeInfo("undefined", "viewer")
fmt.Println(ok) // returns false

hasTypeInfo, ok := HasTypeInfo("document", "undefined")
fmt.Println(ok) // returns false

hasTypeInfo, ok := HasTypeInfo("document", "viewer")
fmt.Println(ok) // returns true
```